### PR TITLE
Feature/perf metrics

### PR DESCRIPTION
**Pull request by ConradoClark**
#### What is the purpose of this pull request?
- Added memory caching on some objects (auth-info from VtexId, MD client entity and app settings)
- Added a prometheus custom metric "safedata_request_total" with "account", "entity" and "statusCode" labels
- Changed app scaling configuration

#### What problem is this solving?
This PR helps with the performance of the app, and allows creating useful dashboards in Grafana to monitor its usage.

#### How should this be manually tested?
You can test this URL on postman:
https://safe--sandboxbrdevbrl.myvtex.com/safedata/CL/search?_fields=id,firstName,lastName
It will return 401 when you're not logged in. It'll return 200 with your data if you're logged in to the store (You are logged in if you have a valid value for the `VtexIdclientAutCookie_sandboxbrdevbrl` Cookie, you can set it manually on postman)

You can check the metrics through [Grafana](http://grafana.ingress.vtex.io/explore?orgId=1&left=%5B%22now-15m%22,%22now%22,%22prometheus%20devs-a%22,%7B%22expr%22:%22sum%20by(entity,%20statusCode)%20(%20rate(safedata_request_total%5B5m%5D)%20)%22,%22requestId%22:%22Q-e6459b47-395f-43b6-8b6b-b67822f98898-0A%22,%22instant%22:false,%22range%22:true%7D%5D). It may take a few seconds to see the results.

#### Screenshots or example usage
Screen Shot 2021-06-18 at 10 27 36

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: node/middlewares/create.ts [hunk 1]
```diff
@@ -163,6 +187,14 @@ async function hasInvalidMatchingDocument({
       } already belongs to a user`,
     })
 
+    ctx.clients.metrics.incrementRequestCounter({
```
Comment (wagnerlduarte) [inline node/middlewares/create.ts:190]: Não daria pra criar uma função com essa chamada de `incrementRequestCounter` para não repetir código e centralizar para manutenções futuras? Pareceu ser a mesma chamada 3 vezes. Faz sentido?

File: node/utils/clientCache.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { LRUCache } from '@vtex/api'
+
+export const DEFAULT_CLIENT_CACHE_MAX_AGE = 1000 * 60 * 5
```
Comment (wagnerlduarte) [inline node/utils/clientCache.ts:3]: O que significa cada número dessa multiplicação? Teria como deixar mais declarativo separando cada um em uma `const` com um nome que informe o que cada um significa. Ex.: `const TEMPO_EM_MINUTOS = 1000`

**Review by wagnerlduarte — APPROVED** [review]

File: node/utils/clientCache.ts (hunk 2 above)
Comment (ConradoClark) [reply node/utils/clientCache.ts]: Boa, corrigido no 3d588c55c134fc8aa538c4427001b693f2d7db25

File: node/middlewares/create.ts (hunk 1 above)
Comment (ConradoClark) [reply node/middlewares/create.ts]: Corrigido no https://github.com/vtex-apps/safedata/pull/8/commits/b2bb05f43dae92e5a808b25cb1ef3d45664c4ab1. As chamadas eram diferentes mas como todas elas estavam essencialmente fazendo a mesma coisa (setando o status, logando se necessário, e disparando as métricas), juntei tudo numa função e substitui as chamadas, ficou mais enxuto.

**Review by wagnerlduarte — APPROVED** [review]

**Pull request merged**